### PR TITLE
AM-3962 PKCE support for IDP - management console

### PR DESCRIPTION
**Pull request by mregulski-gravitee**
## :id: Reference related issue. 
AM-3962

## :pencil2: A description of the changes proposed in the pull request
Management API side counterpart to #4820

### CODE REVIEW

File: gravitee-am-management-api/gravitee-am-management-api-rest/src/main/java/io/gravitee/am/management/handlers/management/api/authentication/controller/LoginController.java [hunk 1]
```diff
@@ -113,7 +129,7 @@ public ModelAndView login(HttpServletRequest request, @RequestParam(value=ORGANI
                 String identityId = identity.getId();
                 SocialAuthenticationProvider socialAuthenticationProvider = (SocialAuthenticationProvider) identityProviderManager.get(identityId);
                 if (socialAuthenticationProvider != null) {
-                    final Maybe<Optional<Request>> maybe = socialAuthenticationProvider.asyncSignInUrl(buildRedirectUri(request, identityId), RandomString.generate()).map(Optional::ofNullable);
+                    final Maybe<Optional<Request>> maybe = socialAuthenticationProvider.asyncSignInUrl(buildRedirectUri(request, identityId), new JWT(), this::processState).map(Optional::ofNullable);
```
Comment (leleueri) [inline gravitee-am-management-api/gravitee-am-management-api-rest/src/main/java/io/gravitee/am/management/handlers/management/api/authentication/controller/LoginController.java:132]: **TODO**: can we at least initialize the iat and exp claim in this state ?

Comment (mregulski-gravitee) [reply gravitee-am-management-api/gravitee-am-management-api-rest/src/main/java/io/gravitee/am/management/handlers/management/api/authentication/controller/LoginController.java:132]: The state JWT we create on the gateway side ([LoginSocialAuthenticationHandler](https://github.com/gravitee-io/gravitee-access-management/blob/b54431c58205fe9f9c2a33d2939a45eccb769754/gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/handler/login/LoginSocialAuthenticationHandler.java#L200)) doesn't have `iat` and `exp` claims. SHould we add them there as well?

Should the expiration time be hardcoded (I'm thinking 5 minutes would be a good value), or do we want users to set it via OIDC provider's configuration?

File: gravitee-am-management-api/gravitee-am-management-api-rest/src/main/java/io/gravitee/am/management/handlers/management/api/authentication/controller/LoginController.java [hunk 2]
```diff
@@ -59,18 +65,16 @@ public class LoginController {
...
-        sMap.put("github-am-idp", "github");
-        sMap.put("google-am-idp", "google");
-        sMap.put("twitter-am-idp", "twitter");
-        sMap.put("facebook-am-idp", "facebook");
-        sMap.put("franceconnect-am-idp", "franceconnect");
-        sMap.put("azure-ad-am-idp", "microsoft");
-        sMap.put("linkedin-am-idp", "linkedin");
-        socialProviderTypes = Collections.unmodifiableMap(sMap);
-    }
+    private static final Duration SOCIAL_IDP_STATE_EXPIRES_AFTER = Duration.ofMinutes(5);
```
Comment (leleueri) [inline gravitee-am-management-api/gravitee-am-management-api-rest/src/main/java/io/gravitee/am/management/handlers/management/api/authentication/controller/LoginController.java:68]: 5 minutes is probably too short if the IDP is configure to use factors like email or mobile app validation. I would define it to 15 minutes and I would make it configurable using a @Value (maybe the key could be `security.stateDuration` in millis)

File: gravitee-am-plugins-handlers/gravitee-am-plugins-handlers-api/src/main/java/io/gravitee/am/plugins/handlers/api/core/impl/ConfigurationFactoryImpl.java [hunk 3]
```diff
@@ -33,7 +34,8 @@ public class ConfigurationFactoryImpl<CONFIGURATION> implements ConfigurationFac
     private final static Logger logger = LoggerFactory.getLogger(ConfigurationFactoryImpl.class);
 
     private final ObjectMapper objectMapper = new ObjectMapper()
-            .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+            .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+            .configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_ENUMS, true);
```
Comment (leleueri) [inline gravitee-am-plugins-handlers/gravitee-am-plugins-handlers-api/src/main/java/io/gravitee/am/plugins/handlers/api/core/impl/ConfigurationFactoryImpl.java:38]: question: why did you add this config ?

**Review by leleueri — APPROVED** [review]

File: gravitee-am-plugins-handlers/gravitee-am-plugins-handlers-api/src/main/java/io/gravitee/am/plugins/handlers/api/core/impl/ConfigurationFactoryImpl.java (hunk 3 above)
Comment (mregulski-gravitee) [reply gravitee-am-plugins-handlers/gravitee-am-plugins-handlers-api/src/main/java/io/gravitee/am/plugins/handlers/api/core/impl/ConfigurationFactoryImpl.java:38]: (resolved on slack)
